src/cascadia/TerminalSettingsModel/FontConfig.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "FontConfig.h"
+#include "FontConfig.g.cpp"
+#include "TerminalSettingsSerializationHelpers.h"
+#include "JsonUtils.h"
+
+using namespace Microsoft::Terminal::Settings::Model;
+using namespace winrt::Microsoft::Terminal::Settings::Model::implementation;
+
+static constexpr std::string_view FontInfoKey{ "font" };
+static constexpr std::string_view FontFaceKey{ "face" };
+static constexpr std::string_view FontSizeKey{ "size" };
+static constexpr std::string_view FontWeightKey{ "weight" };
+static constexpr std::string_view LegacyFontFaceKey{ "fontFace" };
+static constexpr std::string_view LegacyFontSizeKey{ "fontSize" };
+static constexpr std::string_view LegacyFontWeightKey{ "fontWeight" };
+
+winrt::Microsoft::Terminal::Settings::Model::implementation::FontConfig::FontConfig(const winrt::weak_ref<Profile> sourceProfile) :
+    _sourceProfile(sourceProfile)
+{
+}
+
+winrt::com_ptr<FontConfig> FontConfig::CopyFontInfo(const winrt::com_ptr<FontConfig> source, const winrt::weak_ref<Profile> sourceProfile)

```

Comment:
Taking those arguments by value will force the reference count to be incremented for the duration of the call. It's probably a good idea to just take `source` as a pointer or reference as you aren't storing it elsewhere for later.
Especially because the code around `winrt::com_ptr::copy_from` feels a bit non-obvious / roundabout to me.

Reply:
So this was intentionally done this way to maintain consistency with `Profile::CopySettings` and `AppearanceConfig::CopyAppearance`

Reply:
I'm convinced that we'll have to start to clean up such code in the future though.
Isn't it quite seldom that you touch all places simultaneously to clean something up? I don't believe there's (in a rhetoric sense) "any better time to do it than now", as such house-keeping tasks will always be unrelated to any actual task by their nature.